feat(APP-3725): Update Dialog component for 'size' prop (#391)
aragon/gov-ui-kit: CHANGELOG.md

## [Unreleased]

### Added

> - Add `size` prop to `<Dialog /` with `max-w` t-shirt sizing for responsiveness based on viewport

Reviewer: It isn't really based on current viewport, but only on the `size` property
```suggestion
- Add `size` prop to `<Dialog />` core component with `max-w` t-shirt sizing
```
